agent/fw_drupal8.c

```diff
@@ -20,6 +21,60 @@
 
 #define PHP_PACKAGE_NAME "drupal/core"
 
+NR_PHP_WRAPPER(nr_drupal_exception) {
+  int priority = nr_php_error_get_priority(E_ERROR);
+  zval* event = NULL;
+  zval* exception = NULL;
+
+  /* Warning avoidance */
+  (void)wraprec;
+
+  NR_PHP_WRAPPER_REQUIRE_FRAMEWORK(NR_FW_DRUPAL8);
+
+  if (NR_SUCCESS != nr_txn_record_error_worthy(NRPRG(txn), priority)) {
+    NR_PHP_WRAPPER_CALL;
+    goto end;
+  }
+
+  /* Get the event that was given. */
+  event = nr_php_arg_get(1, NR_EXECUTE_ORIG_ARGS TSRMLS_CC);
+
+  /* Call the original function. */
+  NR_PHP_WRAPPER_CALL;
+
+  if (0 == nr_php_is_zval_valid_object(event)) {
+    nrl_verbosedebug(NRL_TXN,
+                     "Drupal: ExceptionSubscriber::onException() does not "
+                     "have an `event` parameter");
+    goto end;
+  }
+
+  /*
+   * Get the exception from the event.
+   */
+  exception = nr_php_call(event, "getThrowable");
+  if (!nr_php_is_zval_valid_object(exception)) {
+    exception = nr_php_call(event, "getException");
+  }
+
+  if (!nr_php_is_zval_valid_object(exception)) {
+    nrl_verbosedebug(NRL_TXN, "Drupal: getException() returned a non-object");
+    goto end;
+  }
```

Comment:
It's there to avoid having to call `nr_php_error_record_exception`, which avoids having to call `nr_php_error_zval_is_exception`, which avoids having to call `nr_php_object_instaceof_class`, then finally call `nr_php_is_zval_valid_object` if it's a non-object entirely and instead just release the `event` and `exception` directly.
Depending on how often this will trigger, it's either more efficient to keep this check (if we expect it to fail often enough) or more efficient to remove it (if we expect it to pass almost all of the time). I'm not positive which case is more likely, but if the consensus is that this check should pass in the overwhelming amount of cases, I'll remove it.